### PR TITLE
[LoopVectorize][X86][AMDLibm] Add Missing AMD LibM trig vector intrinsics

### PR DESCRIPTION
**Pull request by farzonl**
Adding the following linked to their docs:
- [amd_vrs16_acosf](https://github.com/amd/aocl-libm-ose/blob/9c0b67293ba01e509a6308247d82a8f1adfbbc67/scripts/libalm.def#L221)
- [amd_vrd2_cosh](https://github.com/amd/aocl-libm-ose/blob/9c0b67293ba01e509a6308247d82a8f1adfbbc67/scripts/libalm.def#L124)
- [amd_vrs16_tanhf](https://github.com/amd/aocl-libm-ose/blob/9c0b67293ba01e509a6308247d82a8f1adfbbc67/scripts/libalm.def#L224)

### CODE REVIEW

File: llvm/include/llvm/Analysis/VecFuncs.def [hunk 1]
```diff
@@ -1371,7 +1371,9 @@ TLI_DEFINE_VECFUNC("llvm.asin.f32", "amd_vrs16_asinf", FIXED(16), NOMASK, "_ZGV_
 
 TLI_DEFINE_VECFUNC("acosf", "amd_vrs4_acosf", FIXED(4), NOMASK, "_ZGV_LLVM_N4v")
 TLI_DEFINE_VECFUNC("acosf", "amd_vrs8_acosf", FIXED(8), NOMASK, "_ZGV_LLVM_N8v")
+TLI_DEFINE_VECFUNC("acosf", "amd_vrs16_acosf", FIXED(16), NOMASK, "_ZGV_LLVM_N16v")
```
Comment (farzonl) [inline llvm/include/llvm/Analysis/VecFuncs.def:1374]: @rohitaggarwal007 git blame shows you added the amd vector lib support: https://github.com/llvm/llvm-project/pull/78560. Can you review this change?

Comment (farzonl) [reply llvm/include/llvm/Analysis/VecFuncs.def:1374]: @phoebewang gentle ping. Could you take a look at this PR? Also @RKSimon you commented on the last PR (#78560) could you take a look at this PR as well?

**Review by phoebewang — APPROVED** [review]
I'm not familiar with AMDLibm, but I don't see problem in the patch either. So I'll give a green light :)

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**